extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -155,8 +155,15 @@ export class WorkspaceService implements IWorkspaceService {
 		return projectTypes;
 	}
 
-	getProjectsInWorkspace(): vscode.Uri[] {
-		return vscode.workspace.workspaceFile ? this.getWorkspaceConfigurationValue<string[]>(ProjectsConfigurationName).map(project => this.toUri(project)) : [];
+	getProjectsInWorkspace(ext?: string): vscode.Uri[] {
+		let projects = vscode.workspace.workspaceFile ? this.getWorkspaceConfigurationValue<string[]>(ProjectsConfigurationName).map(project => this.toUri(project)) : [];
+
+		// filter by specified extension
+		if (ext) {
+			projects = projects.filter(p => p.fsPath.endsWith(ext));
```

Comment:
p.fsPath.toLowerCase().endsWith(ext.toLowerCase())

Reply:
the OS community at large should just ban all non-lowercased paths. :D

Reply:
Just an FYI, it should be `toUpperCase`, not `toLowerCase`.

https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1308